File: TaskManagement.WebUI/src/app/services/task-comment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import CreateNewCommentRequest from '../data/requests/task-comment/create-comment.request';
import UpdateCommentRequest from '../data/requests/task-comment/update-comment.request';

@Injectable({
  providedIn: 'root',
})
export class TaskCommentService {
  baseUrl: string = 'http://localhost:5000';

  constructor(private httpClient: HttpClient) {}

  getById(id: string) {
    this.httpClient
      .get(`${this.baseUrl}/taskComment/getById/${id}`)
      .subscribe((result) => {
        console.warn(result);
      });
  }

  getAll() {
    this.httpClient
      .get(`${this.baseUrl}/taskComment/getAll`)
      .subscribe((result) => {
        console.warn(result);
      });
  }

  create(payload: CreateNewCommentRequest) {
    this.httpClient
      .post<CreateNewCommentRequest>(
        `${this.baseUrl}/taskComment/create`,
        payload
      )
      .subscribe((result) => {
        console.warn(result);
      });
  }

  update(payload: UpdateCommentRequest) {
    this.httpClient
      .put<UpdateCommentRequest>(`${this.baseUrl}/taskComment/create`, payload)
      .subscribe((result) => {
        console.warn(result);
      });
  }

  remove(taskId: string) {
    this.httpClient
      .delete(`${this.baseUrl}/taskComment/remove/${taskId}`)
      .subscribe((result) => {
        console.warn(result);
      });
  }
}

File: TaskManagement.Application/Services/TaskCommentService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using TaskManagement.Application.Models.Requests.TaskComment;
using TaskManagement.Application.Models.Responses.TaskComment;
using TaskManagement.Application.Services.Abstracts;
using TaskManagement.Domain.Repositories;

namespace TaskManagement.Application.Services
{
    public class TaskCommentService : ITaskCommentService
    {
        private readonly ITaskCommentRepository _taskCommentRepository;
        public TaskCommentService(ITaskCommentRepository taskCommentRepository)
        {
            _taskCommentRepository = taskCommentRepository;
        }

        public async Task<List<TaskCommentDetailsModel>> GetCommentsByTaskId(Guid taskId)
        {
            var comments = await _taskCommentRepository.GetAll();

            var results = new List<TaskCommentDetailsModel>();

            foreach (var comment in comments)
            {
                var taskCommentDetailsModel = new TaskCommentDetailsModel(comment.TaskId, comment.Id, comment.Comment, comment.CommentType, comment.DateAdded, comment.ReminderDate);
                results.Add(taskCommentDetailsModel);
            }

            return results;
        }
        public async Task<Domain.Entities.TaskComment> Create(CreateNewCommentRequest request)
        {
            var taskComment = new TaskManagement.Domain.Entities.TaskComment(request.TaskId, request.Comment, request.CommentType, request.ReminderDate);

            await _taskCommentRepository.Insert(taskComment);

            return taskComment;
        }

        public async Task<Domain.Entities.TaskComment> Update(UpdateCommentRequest request)
        {
            throw new NotImplementedException();
        }

        public async Task<Domain.Entities.TaskComment> Remove(Guid taskCommentId)
        {
            throw new NotImplementedException();
        }
    }
}
